walnuts_cpp/CMakeLists.txt

```diff
@@ -6,6 +6,7 @@ project(walnuts
 )
 
 include(FetchContent)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
```

Comment:
```suggestion
option(WALNUTS_EXPORT_COMPILE_COMMANDS "Generate a compile_commands.json file containing the exact compiler calls for all translation units of the project in machine-readable form." OFF)
if (WALNUTS_EXPORT_COMPILE_COMMANDS)
  set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
endif()
```

Reply:
Is there a reason to not enable this by default? The resulting file is like 1kb and takes almost no time to generate

Reply:
If someone does not use it I don't see a reason to enable it by default.